bokehjs/src/coffee/core/bokeh_events.coffee

```diff
@@ -0,0 +1,170 @@
+# Granular event classes for use with the event system
+import {logger} from "./logging"
+
+
+export class Event
+
+  @_event_classes = {}
+
+  constructor : (options) ->
```

Comment:
You could use `options={}` so that you don't have to `cls({})` every time you construct an event.